Geometry/CaloEventSetup/interface/CaloGeometryLoader.icc

```diff
@@ -98,17 +102,78 @@ void CaloGeometryLoader<T, D>::makeGeometry(const D* cpv,
         nullptr == at ? ht3d
                       : (nullptr == globalT ? at->transform() * ht3d : at->transform() * globalT->transform() * ht3d));
 
-    fillGeom(geom, parameters, atr, id);
+    fillGeom(geom, parameters, atr, id, k_ScaleFromDDDtoGeant);
+  }
+
+  assert(counter <= T::k_NumberOfCellsForCorners);
+
+  geom->initializeParms();
+}
+
+template <class T>
+void CaloGeometryLoader<T>::makeGeometry(const cms::DDCompactView* cpv,
+                             		 T* geom,
+                                         const Alignments* alignments,
+                                         const Alignments* globals) {
+  cms::DDFilteredView fv(cpv->detector(), cpv->detector()->worldVolume());
+
+  fillNamedParams(fv, geom);
+
+  geom->allocateCorners(T::k_NumberOfCellsForCorners);
+  geom->allocatePar(T::k_NumberOfParametersPerShape * T::k_NumberOfShapes, T::k_NumberOfParametersPerShape);
+
+  std::string attribute = "ReadOutName";
+  cms::DDSpecParRefs ref;
+  const cms::DDSpecParRegistry& mypar = cpv->specpars();
+  mypar.filter(ref, attribute, T::hitString());
+  fv.mergedSpecifics(ref);
+
+  unsigned int counter(0);
+  while (fv.firstChild()) {
+    if (boost::starts_with(fv.name(), "EA")) {
+      continue;
+    }
+    ++counter;
+
+    const ParmVec& parameters(fv.parameters());
+
+    DD3Vector x, y, z;
+    fv.rotation().GetComponents(x, y, z);
+    const CLHEP::HepRep3x3 temp(x.X(), y.X(), z.X(), x.Y(), y.Y(), z.Y(), x.Z(), y.Z(), z.Z());
+    const CLHEP::HepRotation hr(temp);
+    const CLHEP::Hep3Vector h3v(fv.translation().X(), fv.translation().Y(), fv.translation().Z());
```

Comment:
We would like to replace CLHEP matrices and vectors with their ROOT equivalents. New code should use the ROOT versions. See: https://hypernews.cern.ch/HyperNews/CMS/get/swDevelopment/3549.html

Reply:
Yes I understand that. We shall do that in one go. We have to handle several classes in CaloGeometry, Ecal, Hcal, HGCal geometry at the same time. I have it on my to-do list. I shall do that later